src/Models/AZ3166Device.ts

```diff
@@ -90,57 +88,71 @@ export class AZ3166Device extends ArduinoDeviceBase {
     const az3166 = boardProvider.find({id: AZ3166Device._boardId});
     return az3166;
   }
+  
+  async compile(): Promise<boolean> {
+    await super.compile();
+    await this.generateBinFile();
+    return true;
+  }
 
-  get version() {
-    // Currently hard code here: version 1.6.1
-    return '1.6.1';
-    // const packageRootPath = this.getArduinoPackagePath();
-    // let version = '0.0.1';
+  private async generateBinFile(): Promise<boolean> {
+    try {
+      if (!fs.existsSync(this.outputPath)) {
+        throw Error(`Output path does not exist.`);

```

Comment:
>`Output path does not exist [](start = 20, length = 27)

Also provide the path for better troubleshooting

Reply:
Done.